pkg/model/appconfig.go

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright (c) 2023 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"path/filepath"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	bsv1alpha1 "redhat-developer/red-hat-developer-hub-operator/api/v1alpha1"
+	"redhat-developer/red-hat-developer-hub-operator/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AppConfigFactory struct{}
+
+// factory method to create App Config object
+func (f AppConfigFactory) newBackstageObject() RuntimeObject {
+	return &AppConfig{MountPath: defaultMountDir}
+}
+
+// structure containing ConfigMap where keys are Backstage ConfigApp file names and vaues are contents of the files
+// Mount path is a patch to the follder to place the files to
+type AppConfig struct {
+	ConfigMap *corev1.ConfigMap
+	MountPath string
+	Key       string
+}
+
+func init() {
```

Comment:
Not sure I understand your point, as you can see it is needed to register factories for runtime objects.
They are not "everywhere" but on the objects which have to be initialized :).
It can be implemented other way, for example hardcoding it, I choose this one.

Reply:
BTW, these objects are mentioned in the design doc as a part of configuration and will be documented more, so if the purpose is to learn how it work - it should be pretty doable :)